Makefile

```diff
@@ -114,7 +114,7 @@ goimports:
 
 .PHONY: staticcheck
 staticcheck:
-	$(STATICCHECK) ./...
+	$(STATICCHECK) -checks "inherit,-SA1019" ./...
```

Comment:
I added this to bypass [SA1019](https://staticcheck.io/docs/checks#SA1019) because the following failure after dependency upgrade:
```
internal/testpb/test.pb.go:42:9: xxx_messageInfo_FooRequest.Unmarshal is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
internal/testpb/test.pb.go:45:9: xxx_messageInfo_FooRequest.Marshal is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
internal/testpb/test.pb.go:48:2: xxx_messageInfo_FooRequest.Merge is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
internal/testpb/test.pb.go:51:9: xxx_messageInfo_FooRequest.Size is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
internal/testpb/test.pb.go:54:2: xxx_messageInfo_FooRequest.DiscardUnknown is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
internal/testpb/test.pb.go:57:32: proto.InternalMessageInfo is deprecated: Do not use; this type existed for intenal-use only.  (SA1019)
internal/testpb/test.pb.go:87:9: xxx_messageInfo_FooResponse.Unmarshal is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
internal/testpb/test.pb.go:90:9: xxx_messageInfo_FooResponse.Marshal is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
internal/testpb/test.pb.go:93:2: xxx_messageInfo_FooResponse.Merge is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
internal/testpb/test.pb.go:96:9: xxx_messageInfo_FooResponse.Size is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
internal/testpb/test.pb.go:99:2: xxx_messageInfo_FooResponse.DiscardUnknown is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
internal/testpb/test.pb.go:102:33: proto.InternalMessageInfo is deprecated: Do not use; this type existed for intenal-use only.  (SA1019)
internal/testpb/test.pb.go:105:2: proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
internal/testpb/test.pb.go:106:2: proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
internal/testpb/test.pb.go:109:15: proto.RegisterFile is deprecated: Use protoregistry.GlobalFiles.RegisterFile instead.  (SA1019)
metrics.go:27:2: package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
metrics_batcher.go:26:2: package cloud.google.com/go/monitoring/apiv3 is deprecated: Please use cloud.google.com/go/monitoring/apiv3/v2.  (SA1019)
metrics_batcher_test.go:23:2: package cloud.google.com/go/monitoring/apiv3 is deprecated: Please use cloud.google.com/go/monitoring/apiv3/v2.  (SA1019)
metrics_proto_api_test.go:27:2: package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
metrics_test.go:23:2: package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
metrics_utils_test.go:24:2: package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
stats.go:34:2: package cloud.google.com/go/monitoring/apiv3 is deprecated: Please use cloud.google.com/go/monitoring/apiv3/v2.  (SA1019)
stats_test.go:23:2: package cloud.google.com/go/monitoring/apiv3 is deprecated: Please use cloud.google.com/go/monitoring/apiv3/v2.  (SA1019)
trace.go:25:2: package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
trace_proto_test.go:27:2: package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
```
Please let me know if this should be an issue or be fix as part of the PR.

Reply:
I'd prefer not disabling the deprecated notices.  I might consider disabling specific files with //lint directives if the upgrade is problematic for some reason, but only if it isn't possible to use the new package easily.

Reply:
I think I have fixed the staticcheck errors. I had to make quite a bit of changes. I did test it by deploying it into Knative cluster just to be sure both metrics and traces are being exported correctly. Please let me know what you think. Thanks again!